### DISCUSSION TITLE
Why `:href` starts with `/pages/` for my dynamic routes

### DISCUSSION THREAD

**Discussion by sharpchen** [vuejs/vitepress#3213] [Q&A]
## Project Structure

My project root is `docs`, this is how I organize my project.
```
docs
|
+---.vitepress
|
+---components
|   +---DocumentStructure.vue
|
+---data
|   +---Document.data.ts
|
+---document
|   +---// some common markdown files
|
+---pages
|   +---[docRoute].md
|   +---[docRoute].paths.ts
|
\---theme
```

## My component and `[docRoute].md`

What I am doing is basically get paths relative to project root for those markdown files, as routes.
But the actual links start with `/pages/` where the `[docRoute].md` is placed.
- **_Actual_**
`localhost:port/pages/../xxx.md`
-  **_Expected_**
`localhost:port/../xxx.md` is expexted. 

I've set base as `/` in `config.mts`, shouldn't it be the same value as `withBase` returns?

```vue
<template>
    <div v-if="currentDoc">
        <table v-if="currentDoc.content.chapters.length > 0"></table>
        <table v-else>
            <div v-for="file in currentDoc.content.filesRelativeToProjectRoot">
                <a :href="withBase(file)">{{ withBase(file) }}</a>
            </div>
        </table>
    </div>
</template>

<script setup lang="ts">
    import { data as documentInfos } from '../data/Document.data';
    import { withBase } from 'vitepress';
    const props = defineProps<{ docRoute: string }>();
    const currentDoc = documentInfos.filter(x => x.docRoute === props.docRoute)[0];
</script>
```

```vue
---
page: true
aside: false
---

<DocumentStructure :doc-route="params.docRoute"/>

<script lang="ts" setup>
    import DocumentStructure from'../components/DocumentStructure.vue';
    import { useData } from 'vitepress';
    const { params } = useData();
</script>
```

**Comment by brc-dd** [comment]
Because you've kept them inside a pages/ directory? You're probably confused with how other frameworks need a pages directory. VitePress doesn't need one, move those files outside that it would work fine.

**Reply by sharpchen** [reply]
It's understandable registered paths of `[docName].md` starts with `/pages/`, but those generated(by my self) links for anchors also starts with `/pages/` which is confusing, is it by design? I don't see such behavior described in vitepress document
